### PR TITLE
Export @required from src/widgets/framework.dart

### PR DESCRIPTION
**Pull request by scheglov**
https://github.com/flutter/flutter/issues/17128

### CODE REVIEW

**Comment by xster** [comment]
This seems totally reasonable. I would actually just copy out the meta part from https://github.com/flutter/flutter/blob/master/packages/flutter/lib/foundation.dart completely. Though I don't have the historical context. 

I'd also clean up the various importers of foundation from widget importers in the repo (don't know if there's a lint for it).

**Comment by scheglov** [comment]
PTAL

I update the PR to export the whole `package:meta` and removed `package:flutter/foundation.dart` where it is not required anymore.

File: packages/flutter/lib/src/widgets/framework.dart [hunk 1]
```diff
@@ -14,7 +14,14 @@ import 'focus_manager.dart';
 
 export 'dart:ui' show hashValues, hashList;
 
-export 'package:flutter/foundation.dart' show required;
+export 'package:meta/meta.dart' show
```
Comment (xster) [inline packages/flutter/lib/src/widgets/framework.dart]: I think it's cleaner if we `show`ed from foundation instead

**Review by xster — APPROVED** [review]
Super awesome. Thanks! 

Please get Ian's LGTM too.

**Comment by Hixie** [comment]
seems reasonable. LGTM.

**Comment by xqwzts** [comment]
While not a breaking change I think this is definitely a mailinglist-worthy one, developers can now clean up their code to remove framework imports too :)

**Comment by Hixie** [comment]
@xqwzts Please don't hesitate to write such an e-mail!

**Pull request merged**